src/main/java/seedu/address/logic/commands/DependencyCommand.java

```diff
@@ -46,27 +47,36 @@ public CommandResult execute(Model model, CommandHistory history) throws Command
                 || dependeeIndex.getZeroBased() >= lastShownList.size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
         }
-        //Checking if dependant or dependee already contains dependency
+        //Creating a different task depending on whether the task dependency currently exists
         Task taskDependant = lastShownList.get(dependantIndex.getZeroBased());
         Task taskDependee = lastShownList.get(dependeeIndex.getZeroBased());
-        if (taskDependant.getDependency().containsDependency(taskDependee)) {
-            throw new CommandException(MESSAGE_ALREADY_DEPENDANT);
-        }
-        //Checking if introducing dependency will create a cyclic dependency
-        Task updatedTask = createDependantTask(taskDependant, taskDependee);
-        DependencyGraph dg = new DependencyGraph(model.getTaskManager().getTaskList());
-        if (dg.checkCyclicDependency(updatedTask)) {
-            throw new CommandException(MESSAGE_CYCLIC_DEPENDENCY);
+
+        Task updatedTask;
+        String message;
+        //Toggle dependency
+        if (taskDependant.isDependentOn(taskDependee)) {
+            //If taskDependant is already dependant on dependee, remove dependency
+            updatedTask = createUndependantTask(taskDependant, taskDependee);
+            message = MESSAGE_REMOVE_SUCCESS;
+        } else {
+            //If taskDependant is not dependent on dependee, add dependency
+            updatedTask = createDependantTask(taskDependant, taskDependee);
+            DependencyGraph dg = new DependencyGraph(model.getTaskManager().getTaskList());
+            //Checking if introducing dependency will create a cyclic dependency
+            if (dg.checkCyclicDependency(updatedTask)) {
```

Comment:
as checkCyclicDependency returns a boolean, naming it something like "containsCyclicDependency" of "willContainCylicDependency" would be more readable?

Reply:
Done